java/src/jmri/managers/AbstractProxyManager.java

```diff
@@ -613,6 +615,18 @@ protected void recomputeNamedBeanSet() {
         return Collections.unmodifiableSortedSet(namedBeanSet);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @OverridingMethodsMustInvokeSuper
+    public void setPropertyChangesMuted(String propertyName, boolean muted) {
```

Comment:
- I think it would be good to send the event on muted -> false regardless of which propertyName is provided.  Otherwise somebody is going to track code other property some day and get a confusing surprise.  

 - Is the behavior specified if setPropertyChangeMuted(.. , false) is called twice?

Reply:
> * I think it would be good to send the event on muted -> false regardless of which propertyName is provided.  Otherwise somebody is going to track code other property some day and get a confusing surprise.

I'll have to think about how to do that. A naive approach would be to just always fire a PropertyChangeEvent with null property values whenever setting muted to false. A more sophisticated approach would involve retaining the old value and sending the current value for any property that is muted while it is muted, and then firing the event with the old value and the current value when unmuted a muted property.

> * Is the behavior specified if setPropertyChangeMuted(.. , false) is called twice?

I have not specified any behavior if `setPropertyChangeMuted(..., false)` is called twice. Do you prefer that I define `setPropertyChangeMuted(...)` be idempotent if the parameter equals the current state of muted?

Reply:
re: any property name - that's probably more effort than needed for the concern. How about throwing an IllegalArgumentException so somebody coding it releases it's missing?  Or logging at Error so it fails CI? (Assuming the tests trip it)

re behavior: I think it would be helpful to add a line or two in the Manager Javadoc, something like "resetting to false will fire a property event that represents state after any muted events, but is not necessarily exactly those events; don't expect it to be an exact queue. Setting to false twice may or may not fire the same events."

Reply:
(Sorry, pushed comment too soon). What I'd like to avoid is "Oh, if I just set that false, I can fire events through the system; cool!"

Reply:
> What I'd like to avoid is "Oh, if I just set that false, I can fire events through the system; cool!"

I'm beginning to think that if that is a desired side effect, I need to track the state of the property to prevent

```
m.setPropertyChangeMuted("foo", true);
m.setPropertyChangeMuted("foo", false);
```

from being such a trigger.

Reply:
@bobjacobsen 
> What I'd like to avoid is "Oh, if I just set that false, I can fire events through the system; cool!"

I'm reading the thread, but I don't really understand all the comments. Can you explain what you mean with this?

Reply:
> I'm beginning to think that if that is a desired side effect, I need to track the state of the property

Or, as it's doing now, check for a particular property name that's allowed to be muted, and throw an exception if there's an attempt to mute any other one.

> Can you explain what you mean with this?

People tend to grab and ad-hoc solution at hand in JMRI, particularly new people. There have been multiple cases where, faced with "well, I updated stuff, but it's not updating on the screen", instead of fixing the underlying problem the (new) programmer looks around for something that has the side-effect of firing semi-random events until the screen eventually does update.  (I just took a case of that out of LayoutEditor; it was a pain to fix right) So I prefer to keep event-firing things focused on _just_ what they need to do. But it's not an absolute rule, just a preference for avoiding code smells.  If it's hard to implement in the code, a Javadoc comment is an OK 2nd choice.